### PR TITLE
[Exporter.Instana] Update Instana Transport with exception-handling and a couple of bug fixes.

### PR DESCRIPTION
**Pull request by TravisTroyer**
Fixes [#747](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/747).

## Changes

- Adds `InstanaExporterEventSource` to provide for error logging.
- Adds exception-handling to `Transport` with logging via `InstanaExporterEventSource`.
- Fixes `Transport` request buffering to remove potential for exceeding underlying array capacity.
- Fixes `Transport` to remove potential for lost spans due to buffer length.

### CODE REVIEW

**Comment by zivaninstana** [comment]
I will take a look today

File: src/OpenTelemetry.Exporter.Instana/Implementation/InstanaExporterEventSource.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Threading;
+
+namespace OpenTelemetry.Exporter.Instana.Implementation;
+
+[EventSource(Name = "OpenTelemetry-Exporter-Instana")]
+internal sealed class InstanaExporterEventSource : EventSource
+{
+    public static InstanaExporterEventSource Log = new();
```
[the remark this answers is not in the document]
Comment (TravisTroyer) [reply src/OpenTelemetry.Exporter.Instana/Implementation/InstanaExporterEventSource.cs]: @Kielek Thanks for taking a look.  I just pushed the addition of a private constructor, along with fixing the line endings in that file.

**Review by zivaninstana — APPROVED** [review]
LGTM 👍

**Review by utpilla — APPROVED** [review]

**Comment by utpilla** [comment]
Thanks for your contribution @TravisTroyer! If you would like to request a new release for this package, please follow the steps mentioned here: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md#how-to-request-for-release-of-package

**Pull request merged**